### PR TITLE
enable addon manager

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: liuwei <liuweixa@redhat.com>

### CODE REVIEW

File: manifests/cluster-manager/0000_01_addon.open-cluster-management.io_managedclusteraddons.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  creationTimestamp: null
+  name: managedclusteraddons.addon.open-cluster-management.io
```
Comment (qiujian16) [inline manifests/cluster-manager/0000_01_addon.open-cluster-management.io_managedclusteraddons.crd.yaml:5]: we should also install clustermanagementaddon crd

Comment (skeeey) [reply manifests/cluster-manager/0000_01_addon.open-cluster-management.io_managedclusteraddons.crd.yaml:5]: added

File: hack/init.sh [hunk 2]
```diff
@@ -5,6 +5,7 @@ set -o nounset
 set -o pipefail
 
 HUB_CRD_FILES="./vendor/github.com/open-cluster-management/api/cluster/v1/*.crd.yaml
+./vendor/github.com/open-cluster-management/api/addon/v1alpha1/0000_01_addon.open-cluster-management.io_managedclusteraddons.crd.yaml
```
Comment (qiujian16) [inline hack/init.sh]: *.crd.yaml I think

Comment (skeeey) [reply hack/init.sh]: updated

File: test/e2e/common.go [hunk 3]
```diff
@@ -501,3 +509,62 @@ func (t *Tester) PodLog(podName, nameSpace string, lines int64) (string, error)
...
+
+func (t *Tester) CheckManagedClusterAddOnStatus(managedClusterNamespace, addOnName string) error {
+	addOn, err := t.AddOnClinet.AddonV1alpha1().ManagedClusterAddOns(managedClusterNamespace).Get(context.TODO(), addOnName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if addOn.Status.Conditions == nil {
+		return fmt.Errorf("there is no conditions in addon %v/%v", managedClusterNamespace, addOnName)
+	}
+	cond := meta.FindStatusCondition(addOn.Status.Conditions, "ManagedClusterAddOnConditionAvailable")
```
Comment (qiujian16) [inline test/e2e/common.go]: nit: meta.IsStatusConditionTrue() will be simpler

Comment (skeeey) [reply test/e2e/common.go]: changed

**Comment by skeeey** [comment]
refer to https://github.com/open-cluster-management/backlog/issues/10141

**Pull request merged**